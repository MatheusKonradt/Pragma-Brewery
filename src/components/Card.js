import React from 'react';
import PropTypes from 'prop-types';
import './Card.sass';

class Card extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.classes = {
      root: 'Card',
      inner: 'inner',
    };
    this.themes = {};
  }

  render() {
    return (
      <div className={[this.classes.root, this.props.className].join(' ')} style={this.props.themes.root}>
        <div className={this.classes.inner} style={this.props.themes.inner}>
          {this.props.children}
        </div>
      </div>
    );
  }
}

Card.defaultProps = {
  themes: {},
  className: '',
};

Card.propTypes = {
  themes: PropTypes.object,
  className: PropTypes.string,
};

export default Card;
