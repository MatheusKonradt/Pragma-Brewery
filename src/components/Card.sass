.Card
  width: 100%
  padding: 5px
  box-sizing: border-box

  div.inner
    width: 100%
    height: 100%
    max-width: 100%
    max-height: 100%
    box-sizing: border-box
    padding: 3px
