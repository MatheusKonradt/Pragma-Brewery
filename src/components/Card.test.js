import React from 'react';
import renderer from 'react-test-renderer';
import Card from "./Card";

test('Should match snapshot', () => {
  const component = renderer.create(
    <Card />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should match snapshot with children nodes', () => {
  const component = renderer.create(
    <Card>
      <p>Testing</p>
    </Card>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});