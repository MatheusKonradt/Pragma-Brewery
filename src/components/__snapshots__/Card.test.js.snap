// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  className="Card "
>
  <div
    className="inner"
  />
</div>
`;

exports[`Should match snapshot with children nodes 1`] = `
<div
  className="Card "
>
  <div
    className="inner"
  >
    <p>
      Testing
    </p>
  </div>
</div>
`;
